vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/credentialverifier/types/msomdoc/MsoMdocVerifiableCredential.kt

```
@@ -59,27 +122,77 @@ class MsoMdocVerifiableCredential : VerifiableCredential {
+            return Triple(issuerAuth, issuerSignedNamespaces, false)
+        }
+
+        if (decodedCredential.keys.toString().contains(ISSUER_AUTH) &&
+            decodedCredential.keys.toString().contains(NAME_SPACES)
+        ) {
+            val issuerAuth: Array = getIssuerAuth(decodedCredential)
+            val issuerSignedNamespaces: Map = (decodedCredential[NAME_SPACES]) as Map
+            return Triple(issuerAuth, issuerSignedNamespaces, true)
+        }
+
+        throw RuntimeException("Invalid issuerSigned structure in mDoc")
```

detekt (1.23.8)

[warning] RuntimeException is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.

(detekt.exceptions.TooGenericExceptionThrown)